aws-iot-securityprofile/src/main/java/com/amazonaws/iot/securityprofile/CreateHandler.java

```diff
@@ -0,0 +1,120 @@
+import software.amazon.awssdk.services.iot.model.IotException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.HandlerErrorCode;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.cloudformation.resource.IdentifierUtils;
+
+public class CreateHandler extends BaseHandler<CallbackContext> {
+
+    // Copied value from software.amazon.cloudformation.resource.IdentifierUtils
+    private static final int GENERATED_NAME_MAX_LENGTH = 40;
+
+    private final IotClient iotClient;
+
+    public CreateHandler() {
+        iotClient = IotClient.builder().build();
+    }
+
+    @Override
+    public ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            AmazonWebServicesClientProxy proxy,
+            ResourceHandlerRequest<ResourceModel> request,
+            CallbackContext callbackContext,
+            Logger logger) {
+
+        CreateSecurityProfileRequest createRequest = translateToCreateRequest(request);
+
+        ResourceModel model = request.getDesiredResourceState();
+        if (!StringUtils.isEmpty(model.getSecurityProfileArn())) {
+            logger.log(String.format("Arn is read-only, but the caller passed %s.", model.getSecurityProfileArn()));
+            // Note: this is necessary even though Arn is marked readOnly in the schema.
+            return ProgressEvent.failed(model, callbackContext, HandlerErrorCode.InvalidRequest,
+                    "Arn is a read-only property and cannot be set.");
+        }
+
+        CreateSecurityProfileResponse createResponse;
+        try {
+            createResponse = proxy.injectCredentialsAndInvokeV2(
+                    createRequest, iotClient::createSecurityProfile);
+        } catch (IotException e) {
+            throw Translator.translateIotExceptionToCfn(e);
+        }
+
+        model.setSecurityProfileArn(createResponse.securityProfileArn());
+        logger.log("Created " + createResponse.securityProfileArn());
+
+        // We're letting customers manage Security Profile attachments in the same CFN template,
+        // using the TargetArns field. Thus, we need to make an AttachSecurityProfile call for every target.
+        Set<String> targetArns = model.getTargetArns();
+        if (targetArns != null) {
+            for (String targetArn : targetArns) {
+                attachSecurityProfile(model.getSecurityProfileName(), targetArn, proxy);
+                logger.log("Attached the security profile to " + targetArn);
+            }
+        }
+
+        return ProgressEvent.defaultSuccessHandler(model);
+    }
+
+    private CreateSecurityProfileRequest translateToCreateRequest(
+            ResourceHandlerRequest<ResourceModel> request) {
+
+        ResourceModel model = request.getDesiredResourceState();
+
```

Comment:
We need to add that we are not trying to set arn like we have done for other resources.